lib/InitModelTypes/InitModelTypes.cpp

```diff
@@ -42,53 +40,47 @@ using llvm::BasicBlock;
 using llvm::Function;
 using llvm::Instruction;
 
-using llvm::cast;
 using llvm::dyn_cast;
-using llvm::isa;
-
-using model::Binary;
-using model::QualifiedType;
 
 template<typename T>
 using RPOT = llvm::ReversePostOrderTraversal<T>;
 
-using TypeVector = llvm::SmallVector<QualifiedType, 8>;
-using ModelTypesMap = std::map<const llvm::Value *, const model::QualifiedType>;
+using TypeVector = llvm::SmallVector<model::UpcastableType, 8>;
+using ModelTypesMap = std::map<const llvm::Value *,
+                               const model::UpcastableType>;
 
-/// Map each llvm::Argument of the given llvm::Function to its
-/// QualifiedType in the model.
+/// Map each llvm::Argument of the given llvm::Function to its type in the model
 static void addArgumentsTypes(const llvm::Function &LLVMFunc,
-                              const model::TypeDefinition *Prototype,
-                              const Binary &Model,
+                              const abi::FunctionType::Layout &Layout,
+                              const model::Binary &Model,
                               ModelTypesMap &TypeMap,
                               bool PointersOnly) {
 
-  const auto Layout = abi::FunctionType::Layout::make(*Prototype);
```

Comment:
NOTETOSELF: review

Reply:
If you mean `InitModelTypes.cpp` in general, then @pfez already took a look.